src/latentis/measure/cka.py

```diff
@@ -0,0 +1,190 @@
+import latentis
+from latentis.measure._metrics import Metric
+import torch 
+import math
+from enum import auto
+
+try:
+    # be ready for 3.10 when it drops
+    from enum import StrEnum
+except ImportError:
+    from backports.strenum import StrEnum
+
+from enum import auto
+import torch
+import math
+
+class CKAMode(StrEnum):
+    """
+    Modes for Centered Kernel Alignment (CKA).
+
+    Attributes:
+        LINEAR: linear mode CKA.
+        RBF: Radial Basis Function (RBF) mode CKA.
+    """
+    LINEAR = auto()
+    RBF = auto()
+
+class CKA(Metric):
+    """
+    A class for computing Centered Kernel Alignment (CKA) between two matrices.
+    Paper https://arxiv.org/abs/1905.00414
+
+    This class supports both linear and RBF kernel methods for computing CKA.
+
+    Attributes:
+        mode (CKAMode): The mode of CKA (linear or RBF).
+        device (torch.device): The torch device (e.g., CPU or GPU) to perform calculations on.
+    """
+
+    def __init__(self, mode: CKAMode, device: torch.device = None):
+        """Initialize the CKA instance with a specific mode and torch device."""
+
+        super().__init__(CKA)
+        
+        self.mode = mode 
+        if self.mode == CKAMode.LINEAR:
+            self.hsic = self.linear_HSIC
+        elif self.mode == CKAMode.RBF:
+            self.hsic = self.kernel_HSIC
+        else:
+            raise NotImplementedError(f"No such mode {self.mode}")
+
+        self.device = device
+
+        # to avoid numerical issues in the assertions
+        self.tolerance = 1e-6
+
+    def _forward(self, space1: torch.Tensor, space2: torch.Tensor, sigma=None):
```

Comment:
Why not `forward`?

Reply:
following https://vscode.dev/github/Flegyas/latentis/blob/feature/measures/src/latentis/measure/_metrics.py#L28

Reply:
Oh right 🍻